src/hooks/core/chat/useMoveChatTypeHook.ts

```diff
@@ -0,0 +1,176 @@
+import {v4 as uuid} from 'uuid';
+import {ChannelType} from '../../../../types/repo/ChannelData';
+import useLocalDatabaseHook from '../../../database/hooks/useLocalDatabaseHook';
+import ChannelListMemberSchema from '../../../database/schema/ChannelListMemberSchema';
+import ChannelList from '../../../database/schema/ChannelListSchema';
+import ChatSchema from '../../../database/schema/ChatSchema';
+import UserSchema from '../../../database/schema/UserSchema';
+import {moveChatToSigned, moveChatToAnon} from '../../../service/chat';
+import {DEFAULT_PROFILE_PIC_PATH} from '../../../utils/constants';
+import {getAnonymousChatName, getChatName} from '../../../utils/string/StringUtils';
+import useUserAuthHook from '../auth/useUserAuthHook';
+import {ANONYMOUS} from '../constant';
+import useChatUtilsHook from './useChatUtilsHook';
+import {generateAnonProfileOtherProfile} from '../../../service/anonymousProfile';
+
+type ChannelCategory = 'SIGNED' | 'ANONYMOUS';
+interface MoveToChatChannelParams {
+  targetUserId: string;
+  oldChannelId: string;
+  source: 'userId';
+}
+
+const useMoveChatTypeHook = () => {
+  const {localDb, refresh} = useLocalDatabaseHook();
+  const {signedProfileId} = useUserAuthHook();
+  const {goToMoveChat} = useChatUtilsHook();
+
+  const saveChannelList = async (
+    channel: any,
+    channelCategory: ChannelCategory
+  ): Promise<ChannelList | undefined> => {
+    if (!channel?.members?.length) return Promise.reject(Error('no members'));
+
+    const isAnonymous = channelCategory === ANONYMOUS;
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

It's good to check if the channel has members before proceeding, but rejecting a promise with a string is not a best practice. It would be better to use an instance of Error for better stack trace support.
```diff
-    if (!channel?.members?.length) return Promise.reject(Error('no members'));
+    if (!channel?.members?.length) return Promise.reject(new Error('No members in the channel'));
```



<!-- This is an auto-generated comment by OSS CodeRabbit -->